src/proto/h1/role.rs

```diff
@@ -1306,6 +1345,19 @@ impl Client {
             }
         };
 
+        let encoder = encoder.map(|enc| {
+            if enc.is_chunked() {
+                let allowed_trailer_fields: Vec<HeaderValue> =
+                    headers.get_all(header::TRAILER).iter().cloned().collect();
+
+                if allowed_trailer_fields.len() > 0 {
```

Comment:
`is_empty()` method can be used to check the vec is empty.

Reply:
fixed in 9cc26a59ae18ad81ff32eaf8fc1919695d7bfee4